### PR TITLE
Rpc eth call

### PR DESCRIPTION
**Pull request by coffeepots**
Implements `eth_call`.

### CODE REVIEW

**Comment by coffeepots** [comment]
Note: Requires https://github.com/status-im/nim-json-rpc/pull/43

**Comment by coffeepots** [comment]
There's still some work required to test `eth_call`, which is currently on another branch, however this is bashing into other PRs so needs manual merging before it gets too far out of sync.

File: tests/test_rpc.nim [hunk 1]
```diff
@@ -68,6 +68,13 @@ proc doTests =
   waitFor client.connect("localhost", Port(8545))
 
   suite "Remote Procedure Calls":
+    # TODO: Currently returning 'block not found' when fetching header in p2p, so cannot perform tests
+    test "eth_call":
```
Comment (stefantalpalaru) [inline tests/test_rpc.nim]: This test fails with:

```
Exception message: {"code":-32000,"id":1,"message":"eth_call raised an exception","data":"Parameter \"value\" is not valid as a UInt256 \"0x100\""}
```
Is it supposed to fail like that?

Also, is the comment above supposed to be moved to the next test, "eth_getBalance"? That's the one failing with:

```
Exception message: {"code":-32000,"id":3,"message":"eth_getBalance raised an exception","data":"No block hash for number 0"}
```

Comment (stefantalpalaru) [reply tests/test_rpc.nim]: Great.

**Pull request merged**